src/SFA.DAS.EmployerCommitmentsV2.Web/Controllers/ApprenticeController.cs

```diff
@@ -59,7 +60,28 @@ public async Task<IActionResult> Download(DownloadRequest request)
         {
             var downloadViewModel = await _modelMapper.Map<DownloadViewModel>(request);
 
-            return File(downloadViewModel.Content, Constants.ApprenticesSearch.DownloadContentType, downloadViewModel.Name);
+            return new FileCallbackResult(downloadViewModel.ContentType, async (outputStream, _) =>
+            {
+                var moreData = true;
+                while (moreData)
+                {
+                    using (var stream2 = await downloadViewModel.GetAndCreateContent(downloadViewModel.Request))
+                    {
+                        if (stream2.Length == 0)
+                        {
+                            moreData = false;
+                        }
+
+                        downloadViewModel.Request.PageNumber += 1;
+
+                        stream2.CopyTo(outputStream);
+                    }
+
+                    downloadViewModel.Dispose();

```

Comment:
does this dispose before the next loop is executed? if so how do you use `downloadViewModel` again in the next loop?